MANUAL.adoc

```diff
@@ -0,0 +1,419 @@
+= Gizmo User's Manual
+:preface-title: introduction
+:docname: user-manual
+:toc:
+
+[id=introduction]
+== Introduction
+
+Gizmo 2 is a library whose purpose is to make it relatively easy to generate reasonably efficient class files.
+
+The API is loosely modeled after the https://openjdk.org/jeps/484[JDK class file API], in that elements are modeled using lexical scoping.
+In contrast however, many things which are subtly confusing or difficult are simplified considerably.
+
+This library is an evolution of the earlier Gizmo 1 library with further conceptual simplifications and capabilities.
+While Gizmo 2 is not source- or binary-compatible with Gizmo 1, it resides in a non-conflicting package space, so the two can coexist during a transition period of migrating from 1 to 2.
+
+[id=quick-start]
+== Quick start
+
+To get started quickly, include the `io.quarkus.gizmo:gizmo2` artifact in your project build. The following code blob shows how to get started quickly with generating classes:
+
+."Hello world" example
+[source,java]
+----
+public class HelloWorld {
+    public static void main(String[] args) {
+        Gizmo gizmo = Gizmo.create(ClassOutput.fileWriter(Path.of("output")));
+        gizmo.class_("HelloWorldGenerated", zc -> {
```

Comment:
Why `zc`?

Reply:
I tend to use `cc` myself, but then when I generate a constructor, I cannot use `cc` again, so I use `mc` even in constructors...

Reply:
Yeah I use `cc` for constructors, so then I use `zc` as a mnemonic abbreviation for `clazz creator`, which is a traditional variable name for `Class`-typed things.

Another option is to use `mc` for methods *and* constructors by convention, freeing `cc` for classes.